openedx/features/course_experience/views/course_outline.py

```diff
@@ -30,12 +30,17 @@ class CourseOutlineFragmentView(EdxFragmentView):
     Course outline fragment to be shown in the unified course view.
     """
 
-    def render_to_fragment(self, request, course_id=None, page_context=None, **kwargs):
+    def render_to_fragment(self, request, course_id=None, page_context=None, preview=False, **kwargs):
```

Comment:
I think the requirement is that if the course access is set to `public` the outline should show and the links should be enabled. And if the course access is set to `preview`, the outline should show but the links are not enabled. This code is disabling the links in both cases.

Reply:
Hum, that's strange, it's called with the value status of `allow_preview` and renders correctly in the [sandbox](https://pr18537.sandbox.opencraft.hosting/courses/course-v1:edX+DemoX+Demo_Course/course/).

Reply:
Looking at this again, you are right. :) It will likely be clearer if the parameter was made `enable_anonymous_access` (which would be passed the value of the WaffleFlag) and then in the context the exact criteria was listed:
```
'enable_links': request.user.is_authenticated or (enable_anonymous_access and request.user.is_anonymous() and course.access is 'public')
```